### PR TITLE
fix #16

### LINKED ISSUE #16: Restart the script
I made the build but for different reasons the generation of the image of vagrant was blocked.
So I restarted the script and I had to eliminate the part already done to speed up the procedure.
At wordcamp I saw that the image generated wasn't incomplete because the provision was incomplete.

We need to implement a way to restart from the last change!

### PR DESCRIPTION
**Pull request by Mte90**

### CODE REVIEW

File: run.sh [hunk 1]
```diff
@@ -19,18 +19,24 @@ else
 fi
 counter=$((counter+1))
 
-echo -e "\033[0;32m${counter}/${total} Cleaning up build and vvv folders\033[0m"
-rm -rf build vvv
-mkdir -p build vvv
-counter=$((counter+1))
+if [ ! -d "vvv" ]; then
+    echo -e "\033[0;32m${counter}/${total} Cleaning up build and vvv folders\033[0m"
+    rm -rf build vvv
```
Comment (tomjn) [inline run.sh]: This will fail as it only runs if the `vvv` folder doesn't exist, so there's nothing to remove

File: run.sh [hunk 2]
```diff
@@ -19,18 +19,24 @@ else
...
+    mkdir -p build vvv
+    counter=$((counter+1))
+fi
```
Comment (tomjn) [inline run.sh]: there's no else statement with a `git pull`. In which case if stuff was pulled down, we'd need to reprovision afresh and regenerate the box file

File: run.sh [hunk 3]
```diff
@@ -54,51 +60,75 @@ echo -e "\033[0;32m${counter}/${total} Prepackaging box\033[0m"
 VVV_SKIP_LOGO=true vagrant package --output ../build/vvv-contribute.box
 counter=$((counter+1))
 
-echo -e "\033[0;32m${counter}/${total} Destroying temporary VM\033[0m"
-VVV_SKIP_LOGO=true vagrant destroy --force
-counter=$((counter+1))
+if [ ! -f "../build/vvv-contribute.box" ]; then
```
Comment (tomjn) [inline run.sh]: If a bad box gets generated, or a newer VVV comes out, then the script won't rebuild the box as the file will already be there

**Comment by tomjn** [comment]
Some of this is good, but it's too geared towards the scenario of provisioning only got halfway through. Even in that scenario you still have to reprovision from scratch, and this doesn't do a lot to help with that as downloading files and moving them around is the cheapest part of the process

But more importantly, I have to delete the entire folder and reclone it if I want to create a new VVV contributor day setup, else it'll just generate the same box as the previous time when the script is ran.

Instead, if you need to run provisioning again because it failed the first time, we should fix that instead. If it's speed you're after, then separating out the meta environment into separate chunks would be better.

Otherwise, the middle section of the script would need rewriting to work on the assumption of a persistent VVV install, not a temporary one

File: run.sh [hunk 4]
```diff
@@ -19,18 +19,24 @@ else
...
+fi
 
-echo -e "\033[0;32m${counter}/${total} Grabbing VVV"
-git clone https://github.com/Varying-Vagrant-Vagrants/VVV.git --branch=master vvv
-counter=$((counter+1))
+if [ ! -d "vvv/.git" ]; then
+    echo -e "\033[0;32m${counter}/${total} Grabbing VVV"
+    git clone https://github.com/Varying-Vagrant-Vagrants/VVV.git --branch=master vvv
+    counter=$((counter+1))
+fi
```
Comment (Mte90) [inline run.sh:36]: I can add there a `git pull` as else

File: run.sh (hunk 3 above)
Comment (Mte90) [reply run.sh]: I think that can I can improve the check there based on the filesize as example but usually the issues are not on generating the box but doing the provision.

File: run.sh (hunk 1 above)
Comment (Mte90) [reply run.sh]: I will remove it

**Comment by Mte90** [comment]
I am trying the script so I will check all the cases, I wanted just to be ready to speed up.

File: run.sh (hunk 3 above)
Comment (Mte90) [reply run.sh]: maybe we can check the date, if is old go on but it is the same day of today will ignore that step.

**Comment by Mte90** [comment]
I will wait the merge of https://github.com/Varying-Vagrant-Vagrants/CD-USB-Generator/pull/26 before resolve the conflicts.

**Comment by Mte90** [comment]
aligned :-)

**Comment by Mte90** [comment]
I think that is ready for the merge

**Comment by tomjn** [comment]
I'm not sure, I need to spend some time really thinking through the mechanics of this. It would have been easier if each part was in its own PR rather than all together, e.g. the downloading of the installers is an easy win, but the parts where it provisions VVV and pulls down a copy I need to think through the mechanics as those could get borked. The current `master` has the benefit that if the VVV gets royally f***d it doesn't matter because you run it again and it flushes all the borked stuff out and puts things in fresh

**Comment by tomjn** [comment]
Additionally, a lot of the time spent in the script is in the provisioning of the meta environment. Speeding that up or parallelising steps in it would speed things up immensely for all involved

**Comment by Mte90** [comment]
Maybe we can reject this and do another script that is only for the vagrant part.
`run.sh` script will execute that script and also the rust of stuff so we don't need this kind of checks.
If someone need to update vagrant can use the other one or if has problems can execute it again.

**Pull request closed without merging**